doc/source/whatsnew/v0.21.0.txt

```diff
@@ -144,6 +144,7 @@ Bug Fixes
 ~~~~~~~~~
 
 - Fixes regression in 0.20, :func:`Series.aggregate` and :func:`DataFrame.aggregate` allow dictionaries as return values again (:issue:`16741`)
+- Fixes regression when sorting by multiple columns on a datetime array with NaT values (:issue:`16836`)
```

Comment:
double back-ticks around ``NaT``; say ``datetime64`` dtype (its not an array, rather a Series)